app/models/poll.rb

```diff
@@ -55,9 +62,25 @@ def options_in_rank
     current_state.each.with_index.each_with_object(Hash.new { |h, k| h[k] = [] }) { |(x, i), hash| hash[x] << options.ids[i] }
   end
 
+  def max_voters
+    self[:max_voters] || Float::INFINITY
+  end
+
   private
 
+  def calculate_ranks
+    SchulzeBasic.do(vote_results, vote_results.first.count).ranks
+  end
+
   def ensure_status_is_correct
     draft! if options.empty?
   end
+
+  def max_voters_should_be_number
+    errors.add(:max_voters, 'should be number greater than 0') unless (is_a?(Integer) || Float::INFINITY) && max_voters > 0
```

Comment:
Are you allowing integer field to contain `Float::INFINITY` when written to the DB? Why? Will it work across different databases?

I really like the trick with replacing null with `INFINITY` in reading accessor, but I'd advise to put ordinary `null` in your database — float columns are quirky (there are 32-bit ones, 64-bit ones and then some), and integer coercion can play some nasty tricks on you if you would use integer columns on the DB level. (And not all possible DBs will even report errors in data truncation cases, _I'm looking at you, MySQL!_)